senders/telegram/send.go

```diff
@@ -92,35 +118,108 @@ func (sender *Sender) buildMessage(events moira.NotificationEvents, trigger moir
 	return buffer.String()
 }
 
-func (sender *Sender) getChatUID(username string) (string, error) {
-	var uid string
-	if strings.HasPrefix(username, "%") {
-		uid = "-100" + username[1:]
-	} else {
-		var err error
-		uid, err = sender.DataBase.GetIDByUsername(messenger, username)
-		if err != nil {
-			return "", fmt.Errorf("failed to get username uuid: %s", err.Error())
-		}
+func (sender *Sender) getChat(contactValue string) (*Chat, error) {
+	var chat *Chat
+	var err error
+
+	switch {
+	// for private channel contactValue is transformed to be able to fetch it from telegram
+	case strings.HasPrefix(contactValue, "%"):
+		contactValue = "-100" + contactValue[1:]
+		chat, err = sender.getChatFromTelegram(contactValue)
+	// for public channel contactValue is transformed to be able to fetch it from telegram
+	case strings.HasPrefix(contactValue, "#"):
+		contactValue = "@" + contactValue[1:]
+		chat, err = sender.getChatFromTelegram(contactValue)
+	// for the rest of the cases (private chats, groups, supergroups), Chat data is stored in DB.
+	default:
+		chat, err = sender.getChatFromDb(contactValue)
 	}
-	return uid, nil
+
+	return chat, err
 }
 
-func (sender *Sender) getChat(username string) (*telebot.Chat, error) {
-	uid, err := sender.getChatUID(username)
+func (sender *Sender) getChatFromDb(contactValue string) (*Chat, error) {
+	var err error
+
+	chatRaw, err := sender.DataBase.GetIDByUsername(messenger, contactValue)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get username uuid: %s", err.Error())
 	}
-	chat, err := sender.bot.ChatByID(uid)
+
+	chat := Chat{}
+	err = json.Unmarshal([]byte(chatRaw), &chat)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal chat data %s: %s", chatRaw, err.Error())
+	}
+	return &chat, nil
+}
+
+func (sender *Sender) getChatFromTelegram(username string) (*Chat, error) {
+	var err error
```

Comment:
I think declaring a variable is unnecessary here

Reply:
Done https://github.com/moira-alert/moira/pull/1022/commits/f726d1c03b707e870f9b11cde5ba68ffa42c7427